torch/_dynamo/output_graph.py

```diff
@@ -194,43 +196,56 @@ def update_co_names(self, name):
                 name,
             )
 
-    def register_attr_or_module(self, mod: torch.nn.Module, *names, **options):
-        if is_dynamic_nn_module(mod):
-            return variables.UnspecializedNNModuleVariable(mod, **options)
+    def register_attr_or_module(
+        self, target: Union[torch.nn.Module, torch.Tensor, Any], *names, **options
+    ):
+        if is_dynamic_nn_module(target):
+            return variables.UnspecializedNNModuleVariable(target, **options)
 
         options = dict(options)
         options["guards"] = set(options.get("guards", []))
         source: Source = options.get("source", None)
-        if isinstance(mod, torch.Tensor):
+        if isinstance(target, torch.Tensor):
             if source:
                 options["guards"].add(source.make_guard(GuardBuilder.TENSOR_MATCH))
 
             def wrap_name(module_key):
-                return TensorVariable.create(
+                return wrap_fx_proxy(
                     self,
                     self.create_proxy("get_attr", module_key, tuple(), {}),
-                    example_value=mod,
+                    example_value=target,
                     **options,
                 )
 
-        elif isinstance(mod, torch.nn.Module):
-            assert isinstance(mod, torch.nn.Module)
+        elif isinstance(target, torch.nn.Module):
+            assert isinstance(target, torch.nn.Module)
             options["guards"].add(source.make_guard(GuardBuilder.NN_MODULE))
 
             def wrap_name(module_key):
-                return NNModuleVariable(type(mod), module_key, **options)
+                return NNModuleVariable(type(target), module_key, **options)
+
+        elif isinstance(target, (torch.SymInt, torch.SymFloat)):
```

Comment:
This is for any case where we register a symbolic shape w/ register_attr_or_module, which we do elsewhere in this pr

Reply:
And you will see I commented there that it was weird to do :P

Reply:
Hmm, in those cases, should we just keep inputs as non dynamic?

Reply:
Well I don't really see how you can end up with a SymInt stored on a Python data structure like nn.Module, it seems very questionable

Reply:
sorry, why do you think this is an nn module?

Reply:
So what the heck is the logic at the end of this function:

```
        base = name
        for i in itertools.count():
            if name not in self.nn_modules:
                self.nn_modules[name] = target
                return wrap_name(name)
            name = f"{base}_{i}"
```

SymInt doesn't early return, so you're going to hit this right?

Reply:
a bad piece of  leftover logic for global storage + name from when this was only modules. I'll add another object like self.dyn_shapes

Reply:
I started this change, and its kind of a big annoying one, do you mind if we keep this as is and take it on as a TODO after?

Reply:
(As a datapoint, there's tensors in there too, I think)

Reply:
I'm OK with deferring things to later. I just want to understand what the correct state is.

I still don't understand how we can end up with an untracked symint, that needs to be registered. Untracked tensors/modules make sense, because they can exist in the user environment. But we have to intentionally introduce symints, and if we introduced them, we ought to have tracked them already. This is similar to the conversation about channels_in, and it turned out that they were getting created when we were fake-ifying modules.